pkg/sql/catalog/catpb/privilege.go

```diff
@@ -214,6 +214,9 @@ func NewPublicSchemaPrivilegeDescriptor() *PrivilegeDescriptor {
 func (p *PrivilegeDescriptor) CheckGrantOptions(
 	user username.SQLUsername, privList privilege.List,
 ) bool {
+	if p.Owner() == user {
```

Comment:
it is supposed to inherit here, but this was previously being applied in `func (p *planner) CheckPrivilegeForUser`:
https://github.com/cockroachdb/cockroach/blob/d7cf6d2336085a7b8c45549d2364d196deb9164e/pkg/sql/authorization.go#L148-L150

the `checkRolePredicate` goes up the role hierarchy

i actually only added the `if p.Owner() == user` here because there was a backup test that doesnt have access to the planner, but was checking privileges directly on the descriptor

Reply:
>i actually only added the if p.Owner() == user here because there was a backup test that doesnt have access to the planner, but was checking privileges directly on the descriptor

Which test is this? I'm more inclined to updating the test instead of doing this here.